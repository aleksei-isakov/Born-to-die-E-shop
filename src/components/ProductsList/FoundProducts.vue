<template>
  <p class="found-products">{{ productsQuantityValue }}</p>
</template>

<script>
export default {
  name: 'FoundProducts',

  props: {
    productsQuantity: {
      type: Number,
      required: true,
      default: 0
    }
  },

  computed: {
    productsQuantityValue() {
      return `Found: ${this.productsQuantity}`;
    }
  }
};
</script>

<style lang="scss" scoped>
@import '@/scss/CustomVariables.scss';

.found-products {
  font-weight: bold;
  color: $primary;
  margin: 20px 0;
  text-align: left;
}
</style>
