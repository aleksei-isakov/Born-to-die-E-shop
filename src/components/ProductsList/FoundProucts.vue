<template>
  <span class="found-products">Found: {{ itemsPerPage }}</span>
</template>

<script>
export default {
  props: {
    itemsPerPage: {
      type: Number,
      required: true,
      default: 1
    }
  }
};
</script>

<style lang="scss" scoped>
@import '@/scss/variables.scss';

.found-products {
  font-weight: bold;
  color: $primary;
  margin: 20px 0;
}
</style>
