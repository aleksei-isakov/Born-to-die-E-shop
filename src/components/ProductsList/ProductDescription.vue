<template>
  <div
    :class="[
      { 'product-description_horizontal': isHorizontal },
      'product-description'
    ]"
  >
    <h2 class="product-description__title">{{ title }}</h2>
    <p class="product-description__category">{{ getCategory }}</p>
    <span>{{ getCreatedDate }}</span>
    <br />
    <span>{{ getUpdatedDate }}</span>
    <rating-icon class="product-description__rating-icon" :rating="rating" />
  </div>
</template>

<script>
import { format } from 'date-fns';
import RatingIcon from '../RatingIcon/RatingIcon.vue';

export default {
  name: 'ProductDescription',

  components: { RatingIcon },

  props: {
    title: {
      type: String,
      required: true,
      default: ''
    },

    category: {
      type: String,
      required: true,
      default: ''
    },

    created: {
      type: String,
      required: true,
      default: ''
    },

    updated: {
      type: String,
      default: ''
    },

    rating: {
      type: Number,
      required: true,
      default: 0
    },

    isHorizontal: {
      type: Boolean,
      required: true,
      default: false
    }
  },

  computed: {
    getCreatedDate() {
      return `Created: ${format(new Date(this.created), 'DD.MM.YYYY')}`;
    },

    getUpdatedDate() {
      return this.updated
        ? `Updated: ${format(new Date(this.updated), 'DD.MM.YYYY')}`
        : '';
    },

    getCategory() {
      return this.category ? this.category : 'Category';
    }
  }
};
</script>

<style lang="scss" scoped>
@import '@/scss/CustomVariables.scss';

.product-description {
  color: #8b8a8a;
  text-align: start;
  width: 100%;
  margin: 15px 0;

  &_horizontal {
    margin: 0 15px;

    @media screen and (max-width: $tablet-size) {
      margin: 15px;
    }
  }

  &__category {
    font-weight: bold;
  }

  &__title {
    min-height: 65px;
    color: $primary;
    margin-bottom: 10px;
  }

  &__rating-icon {
    margin: 15px 0;
  }
}
</style>
