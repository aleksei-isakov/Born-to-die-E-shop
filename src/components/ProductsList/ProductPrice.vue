<template>
  <div class="product-price"><slot></slot></div>
</template>

<script>
export default {
  name: 'ProductPrice'
};
</script>

<style lang="scss" scoped>
@import '@/scss/CustomVariables.scss';
.product-price {
  color: black;
  font-size: 1.5rem;
}
.crossed {
  text-decoration: line-through;
  color: $font-color-subtitle;
  font-size: 14px;
  margin-left: 10px;
}
</style>
