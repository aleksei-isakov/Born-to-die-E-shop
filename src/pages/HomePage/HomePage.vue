<template>
  <div>
    <PLPSearchBar />
    <ProductsList :products="productsList" />
  </div>
</template>

<script>
import ProductsList from '@/components/ProductsList/ProductsList.vue';
import PLPSearchBar from '@/components/PLPSearchBar/PLPSearchBar.vue';
import { mapGetters, mapActions } from 'vuex';

export default {
  name: 'HomePage',

  components: {
    ProductsList,
    PLPSearchBar
  },

  computed: {
    ...mapGetters('ProductsModule', ['productsList'])
  },

  async mounted() {
    await this.getProductsList();
  },

  methods: {
    ...mapActions('ProductsModule', ['getProductsList'])
  }
};
</script>
