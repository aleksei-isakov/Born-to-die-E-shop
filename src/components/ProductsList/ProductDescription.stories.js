import ProductDescription from './ProductDescription.vue';

export default {
  /* ðŸ‘‡ The title prop is optional.
   * See https://storybook.js.org/docs/vue/configure/overview#configure-story-loading
   * to learn how to generate automatic titles
   */
  title: 'ProductDescription'
};

const ProductDescriptionTemplate = (args) => ({
  components: { ProductDescription },
  props: Object.keys(args),
  template: '<ProductDescription v-bind="$props" />'
});

export const Default = ProductDescriptionTemplate.bind({});

export const WithAnotherTitle = ProductDescriptionTemplate.bind({});

const unixTimestamp = 1549312452;

// About args https://storybook.js.org/blog/introducing-storybook-args/
// https://storybook.js.org/docs/react/writing-stories/args

Default.args = {
  title: 'Product details title',
  field: 'Product details field',
  created: new Date(unixTimestamp),
  updated: new Date()
};

WithAnotherTitle.args = {
  title: 'Another title',
  field: 'Product details field',
  created: new Date(unixTimestamp),
  updated: new Date()
};
