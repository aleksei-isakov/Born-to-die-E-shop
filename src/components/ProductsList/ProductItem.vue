<template>
  <li class="product-item">
    <img class="product-item__image" :src="getImage" alt="product image" />
    <ProductDescription
      v-if="description"
      :title="description.title"
      :field="description.field"
      :created="description.created"
      :updated="description.updated"
    />
    <div class="product-item__price-container">
      <ProductPrice>{{ getPrice }} $</ProductPrice>
      <button class="md-button md-raised md-primary md-theme-default">
        + ADD TO CART
      </button>
    </div>
  </li>
</template>

<script>
import defaultImage from '@/assets/defaultImage.jpg';
import ProductDescription from './ProductDescription.vue';
import ProductPrice from './ProductPrice.vue';

export default {
  name: 'ProductItem',

  props: {
    image: {
      type: String,
      default: ''
    },

    price: {
      type: Number,
      required: true,
      default: 0
    },

    description: {
      type: String,
      default: ''
    }
  },

  data() {
    return {
      defaultImage: defaultImage
    };
  },

  computed: {
    getImage() {
      return this.image ? product.image : defaultImage;
    },

    getPrice() {
      return this.price.toFixed(1);
    }
  },

  components: {
    ProductDescription,
    ProductPrice
  }
};
</script>

<style scoped>
.product-item {
  width: 20vw;
  min-width: 250px;
  height: fit-content;
  border-radius: 10px;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 20px;
  margin: 0 30px 55px 0;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.2);
}
.product-item__price-container {
  width: 100%;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
}
.product-item__image {
  width: 80%;
}
</style>
