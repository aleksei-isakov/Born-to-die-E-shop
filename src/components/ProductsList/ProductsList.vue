<template>
  <div class="products-list">
    <RecentlyAdded text="Recently added" />
    <ul class="products-list__items">
      <ProductItem
        v-for="product in products"
        :key="product.id"
        :image="product.image"
        :price="product.price"
        :description="product.description"
      />
    </ul>
  </div>
</template>

<script>
import ProductItem from './ProductItem.vue';
import RecentlyAdded from './RecentlyAdded.vue';

export default {
  name: 'ProductsList',

  components: {
    ProductItem,
    RecentlyAdded
  },

  props: {
    products: {
      type: Array,
      default: () => [],
      required: true
    }
  }
};
</script>

<style scoped>
.products-list {
  margin-top: 15%;
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  padding-left: 15%;
}
.products-list__items {
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
  padding: 0;
}
</style>
