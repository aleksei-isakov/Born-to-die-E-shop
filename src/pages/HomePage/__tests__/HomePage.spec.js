import { createLocalVue, shallowMount } from '@vue/test-utils';
import Vuex from 'vuex';
import products from '@/components/ProductsList/productsMock.json';
import HomePage from '../HomePage.vue';
import PlpSearchBar from '@/components/PlpSearchBar/PlpSearchBar.vue';
import ProductsList from '@/components/ProductsList/ProductsList.vue';

let wrapper;
let store;
let actions;
let getters;
const localVue = createLocalVue();

localVue.use(Vuex);

beforeEach(() => {
  getters = {
    productsList: () => []
  };
  actions = {
    getProductsList: jest.fn((getters.productsList = () => products))
  };
  store = new Vuex.Store({
    modules: {
      ProductsModule: {
        actions,
        getters,
        namespaced: true
      }
    }
  });
  wrapper = shallowMount(HomePage, {
    stubs: {
      PlpSearchBar: PlpSearchBar,
      ProductsList: ProductsList
    },
    store,
    localVue
  });
});

afterEach(() => {
  wrapper.destroy();
});

const getProductsList = () => wrapper.find('.products-list');

describe('HomePage', () => {
  it('renders a valid snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should contain products list', () => {
    expect(getProductsList().exists()).toBe(true);
  });
});
