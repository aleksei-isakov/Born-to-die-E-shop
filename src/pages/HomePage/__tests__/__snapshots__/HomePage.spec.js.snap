// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentsPage renders a valid snapshot 1`] = `
<div><img alt="Vue logo" src="../../assets/logo.png">
  <introduction-stub msg="Welcome to Your Vue.js App"></introduction-stub>
</div>
`;
